src/platform.ts

```diff
@@ -162,10 +172,29 @@ export function getAvailablePowerShellExes(
         }
     } else {
         // Handle Linux and macOS case
+        const defaultExePath =  this.getDefaultPowerShellPath(platformDetails);
         paths.push({
-            versionName: "PowerShell Core",
-            exePath: this.getDefaultPowerShellPath(platformDetails),
+            versionName: "PowerShell Core" + (/-preview/.test(defaultExePath) ? " Preview" : ""),
+            exePath: defaultExePath,
         });
+
+        // If defaultExePath is pwsh, check to see if pwsh-preview is installed and if so, make it available.
+        // If the defaultExePath is already pwsh-preview, then pwsh is not installed - nothing to do.
+        if (platformDetails.operatingSystem === OperatingSystem.MacOS) {
+            if ((defaultExePath === macOSExePath) && fs.existsSync(macOSPreviewExePath)) {
+                paths.push({
+                    versionName: "PowerShell Core Preview",
+                    exePath: macOSPreviewExePath,
+                });
+            }
+        } else if (platformDetails.operatingSystem === OperatingSystem.Linux) {
+            if ((defaultExePath === linuxExePath) && fs.existsSync(linuxPreviewExePath)) {
+                paths.push({
+                    versionName: "PowerShell Core Preview",
+                    exePath: linuxPreviewExePath,
+                });
+            }
```

Comment:
I refactored this a bit to define the paths conditionally and then do the logic. I think it's a little DRY-er. Totally, not a blocker but what do you think?

```typescript
let osExePath, osPreviewExePath;
if (platformDetails.operatingSystem === OperatingSystem.MacOS) {
    osExePath = macOSExePath;
    osPreviewExePath = macOSPreviewExePath;
} else if (platformDetails.operatingSystem === OperatingSystem.Linux) {
    osExePath = linuxExePath;
    osPreviewExePath = linuxPreviewExePath;
}

if ((defaultExePath === osExePath) && fs.existsSync(osPreviewExePath)) {
    paths.push({
        versionName: "PowerShell Core Preview",
        exePath: linuxPreviewExePath,
    });
}

```

Reply:
Updated.  Please test on macOS and Linux.